mlir/lib/Dialect/EmitC/Transforms/TypeConversions.cpp

```diff
@@ -37,3 +37,28 @@ void mlir::populateEmitCSizeTTypeConversions(TypeConverter &converter) {
   converter.addTargetMaterialization(materializeAsUnrealizedCast);
   converter.addArgumentMaterialization(materializeAsUnrealizedCast);
 }
+
+/// Get an unsigned data type as wide as \p ty.
+std::optional<Type> mlir::emitc::getUnsignedTypeFor(Type ty) {
+  if (ty.isInteger())
+    return IntegerType::get(ty.getContext(), ty.getIntOrFloatBitWidth(),
+                            IntegerType::SignednessSemantics::Unsigned);
+  if (isa<emitc::PtrDiffTType, emitc::SignedSizeTType>(ty))
+    return emitc::SizeTType::get(ty.getContext());
+  if (isSupportedIntegerType(ty))
+    return ty;
```

Comment:
Why is `ty` unsigned here? Same question below

Reply:
That was a wrong call from mine -- SizeTType wasn't even part of those in `isSupportedIntegerType`. Basically we've got built-in integers and SizeT/SSizeT/PtrDiffT types that are accepted here (built-in index is not). I've explicitly written them down.